### ISSUE TITLE
stats don't match database

### ISSUE THREAD

**Issue by bjmgeek** [hockeypuck/server#11]
I'm using hockeypuck 2.0b4 on Debian unstable on amd64, and I had to recreate my `/var/lib/hockeypuck` directory (including the recon db) due to a failed hard drive.  I've still got the postgresql db, and it shows the correct number of keys:

````
select count(*) from keys;
  count  
---------
 3953468
(1 row)
````
but the stats page shows 
````
Statistics
Total number of keys: 97 
````

Is there a way to make the stats re-read the database?

**Comment by bjmgeek** [comment]
or should I just reload a dump?

**Comment by cmars** [comment]
You should be able to fix this by rebuilding the prefix tree, similar to an `sks pbuild`:

```
hockeypuck-pbuild -config /path/to/hockeypuck.conf
```

**Comment by cmars** [comment]
Once the ptree is rebuilt, remove the old `.ptree.stats` file and restart the hockeypuck service. Hockeypuck will then rebuild the stats file from the prefix tree.

**Comment by cmars** [comment]
Couple of things I need to fix:
1. Add documentation on `hockeypuck-pbuild` command.
2. Remove that stats file at exit for `hockeypuck-load` & `hockeypuck-pbuild`.

**Comment by cmars** [comment]
.ptree.stats will typically be located a directory up from your ptree directory.

**Comment by bjmgeek** [comment]
I tried that and got an error:

hockeypuck@hockeypuck:~$ /usr/bin/hockeypuck-pbuild
-config=/etc/hockeypuck/hockeypuck.conf
WARN[0000] error executing "ALTER TABLE keys ADD CONSTRAINT keys_pk PRIMARY
KEY (rfingerprint);": pq: multiple primary keys for table "keys" are not
allowed
WARN[0000] error executing "ALTER TABLE keys ADD CONSTRAINT keys_md5 UNIQUE
(md5);": pq: relation "keys_md5" already exists
WARN[0000] error executing "CREATE INDEX keys_rfp ON keys(rfingerprint
text_pattern_ops);": pq: relation "keys_rfp" already exists
WARN[0000] error executing "CREATE INDEX keys_ctime ON keys (ctime);": pq:
relation "keys_ctime" already exists
WARN[0000] error executing "CREATE INDEX keys_mtime ON keys (mtime);": pq:
relation "keys_mtime" already exists
WARN[0000] error executing "CREATE INDEX keys_keywords ON keys USING
gin(keywords);": pq: relation "keys_keywords" already exists
WARN[0000] error executing "ALTER TABLE subkeys ADD CONSTRAINT subkeys_pk
PRIMARY KEY (rsubfp);": pq: multiple primary keys for table "subkeys" are
not allowed
WARN[0000] error executing "ALTER TABLE subkeys ADD CONSTRAINT subkeys_fk
FOREIGN KEY (rfingerprint) REFERENCES keys(rfingerprint);": pq: constraint
"subkeys_fk" for relation "subkeys" already exists
WARN[0000] error executing "CREATE INDEX subkeys_rfp ON subkeys(rsubfp
text_pattern_ops);": pq: relation "subkeys_rfp" already exists
[{/home/cmars/hkp/gopath/src/
github.com/hockeypuck/packaging/src/github.com/hockeypuck/server/cmd/hockeypuck-pbuild/main.go:103:
} {/home/cmars/hkp/gopath/src/
github.com/hockeypuck/packaging/src/gopkg.in/hockeypuck/pghkp.v1/storage.go:604:
} {sql: Scan error on column index 0: unsupported driver -> Scan pair:
[]uint8 -> *struct { MD5 string "bson:\"md5\"" }}]

**Comment by bjmgeek** [comment]
Do you mean the .recon.db.stats file?

**Comment by cmars** [comment]
Yes, it'd be `.recon.db.stats` if the prefix tree dir was named `recon.db`.

**Comment by cmars** [comment]
Uh-oh, that looks like a likely bug in hockeypuck-pbuild, need to test it with postgres-json storage.

**Comment by cmars** [comment]
Still need to update docs, reopening.

**Comment by bjmgeek** [comment]
I'm actually still seeing this issue

I've tried rebuilding the PTree, reloading from a dump, etc.  My server has been running for several days, and I don't see any activity on the Stats page except for one key I manually added.  I can see that it's gossiping with my sks server, and my peers (both sks and hockeypuck), but no updates on the stats page

Statistics
Total number of keys: 4102179
Daily Histogram

Day|New Keys|Updated Keys
-----|--------------|-------------------
2016-01-04|4102178|0
2016-01-05|1|1
Hourly Histogram

Hour	|New Keys|Updated Keys
-------|-------------|-------------------
2016-01-05 19|1|0
2016-01-05 20|0|1

This for a server that is still running at the present, so it should have had some activity today.

**Comment by royalterra** [comment]
I have the seam issue with mongodb . Hockeypuck install from Ubuntu repository
hockeypuck-pbuild -config /etc/hockeypuck/hockeypuck.conf
That Start ADD keys 
INFO[1186] 275000 keys added                            
INFO[1210] 280000 keys added                            
INFO[1232] 285000 keys added
After all that is end I execute:
rm /var/lib/hockeypuck/.recon.db.stats
service hockeypuck restart

And now is worse total number of keys 1
